gym_wrapper/base_gym_env.py

```diff
@@ -130,20 +180,42 @@ def __init__(
 
         self.observation_space = gym.spaces.Dict(spaces)
 
+        self.game.init()
+
     def step(self, action):
         assert self.action_space.contains(action), f"{action!r} ({type(action)}) invalid"
         assert self.state is not None, "Call `reset` before using `step` method."
 
-        # convert action to vizdoom action space (one hot)
-        act = [0 for _ in range(self.action_space.n)]
-        act[action] = 1
-
+        act = self.__build_action(action)
         reward = self.game.make_action(act, self.frame_skip)
         self.state = self.game.get_state()
         done = self.game.is_episode_finished()
 
         return self.__collect_observations(), reward, done, {}
 
+    def __parse_binary_buttons(self, act, action):
+        if self.binary_buttons != 0:
+            if self.delta_buttons != 0:
+                action = action["binary"]
+
+            if self.max_buttons_pressed == 1:
+                act[action + self.delta_buttons] = 1
+            else:
+                for button in action:
+                    act[button + self.delta_buttons] = 1
+
+    def __parse_delta_buttons(self, act, action):
+        if self.delta_buttons != 0:
+            for delta_idx in range(self.delta_buttons):
+                act[delta_idx] = action["delta_" + str(delta_idx)][0]
+
+    def __build_action(self, action):
+        # convert action to vizdoom action space (one hot)
```

Comment:
Nitpick, but the action space is not one hot here if it is dictionary or something :). "One hot" referes to a vector with all but one entry being zeros, and the chosen one being one.

Reply:
yeah it's more multi-hot encoded or just the "encoded" action

Reply:
Maybe just "environment action" (I personally use terms "agent action" [what comes out from agent network] and "environment action" [what goes into step]), but "encoded" works too :)